tests/unit_tests/scheduler/test_scheduler.py

```diff
@@ -334,18 +334,27 @@ async def kill():
 
 
 @pytest.mark.timeout(6)
-async def test_that_job_does_not_retry_when_killed(realization, mock_driver):
+async def test_that_job_does_not_retry_when_killed_by_scheduler(
```

Comment:
`killed_by_cancellation` ?

Reply:
"killed_by_cancellation" sounds more like we have cancelled the job task in the test, but rather we are using Schedulers API to kill the job tasks (that Scheduler is using cancellation to do the killing is an implementation detail) (there is another `test_cancel` that steps into internals of the scheduler and issues cancellation on its own).